### PR TITLE
ENG-1081: Confluence integration

### PR DESCRIPTION
**Pull request by daabr**
Turns out Confluence and Jira are not at all alike as one would assume, and that Confluence's APIs are... Not as mature as one would expect.

Specific notes:

- Webhook creation via REST in Confluence **Cloud** is hacky
- Works only with API tokens and not OAuth 2.0 (again, only in Cloud)
- These webhooks don't support secrets for callback signature verification
- No event type ID in callbacks, need to use a separate webhook URL per event category (created, updated, removed, etc.) - focusing only on "created" for now

### CODE REVIEW

File: integrations/atlassian/confluence/event.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+	// Parse some of the metadata in the Atlassian event's JSON content.
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		l.Warn("Failed to read content of incoming Atlassian event", zap.Error(err))
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return
+	}
+
+	var atlassianEvent map[string]any
```
Comment (pashafateev) [inline integrations/atlassian/confluence/event.go:72]: Why is this `map[string]any` and not something more specific? Is that due to not having much information about the events to begin with or is this common practice?

**Review by pashafateev — APPROVED** [review]

File: integrations/atlassian/confluence/event.go (hunk 1 above)
Comment (daabr) [reply integrations/atlassian/confluence/event.go:72]: Exactly. `map[string]any` is the most generic way to unmarshal JSON text into a Go data structure, without knowing anything about it (literally "a mapping of string keys to whatever"). In the case of Atlassian events - their schema varies so wildly that it doesn't make sense to enforce any structure on them between the point of receiving them (our webhook) and dispatching them to session workflows.

File: integrations/atlassian/confluence/vars.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+var (
+	baseURL = sdktypes.NewSymbol("BaseURL")
+	token   = sdktypes.NewSymbol("Token")
+	email   = sdktypes.NewSymbol("Email")
+
+	oauthAccessToken = sdktypes.NewSymbol("oauth_AccessToken")
```
Comment (itayd) [inline integrations/atlassian/confluence/vars.go:12]: i think using the struct approach would have been nicer here

File: integrations/atlassian/confluence/webhooks.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		l.Warn("Failed to read Confluence webhooks list response", zap.Error(err))
+		return 0, false
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		l.Warn("Unexpected response to Confluence webhooks list request",
+			zap.Int("status", resp.StatusCode),
+			zap.ByteString("body", body),
```
Comment (itayd) [inline integrations/atlassian/confluence/webhooks.go:78]: you sure you want to spit out the body here?

File: integrations/atlassian/confluence/webhooks.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+		)
+		return 0, false
+	}
+
+	var list []webhook
+	if err := json.Unmarshal(body, &list); err != nil {
```
Comment (itayd) [inline integrations/atlassian/confluence/webhooks.go:84]: you could use a decoder and skip the ReadAll

**Review by itayd — COMMENTED** [review]
only nits

**Pull request merged**